frontend/src/liboperator/operator/operator.go

```diff
@@ -513,7 +513,6 @@ func adbProxy(w http.ResponseWriter, r *http.Request, pool *DevicePool) {
 		log.Print("Error while connect to ADB: ", err)
 		return
 	}
-	defer tcpConn.Close()
```

Comment:
Suggest to rather keep the two defer calls: `defer tcpConn.Close()` and `defer wsWrapper.Close()` as they are and remove `wsWrapper.Close()` within the goroutine. I prefer this approach as it's a common Golang idiom when working with `Closeable` instances.

Reply:
The `defer tcpConn.Close()` should stay, but the one for `wsWrapper` needs to go in the goroutine. The reason is that the copy from `wsWrapper` to `tcpConn` can finish before the goroutine completes writing to `wsWrapper`, which will lead to `wsWrapper` being closed in the middle of a write operation. To avoid this, sockets should be closed in the goroutine they're written to. Alternatively, you can keep the defer for `wsWrapper` but must ensure the function doesn't return until the goroutine finishes.

Reply:
Changed to keep tcpConn.Close() and close wsWrapper inside goroutine.